tests/instances/ok/boolean_checks.txt

```diff
@@ -0,0 +1,13 @@
+#TIMEHORIZON T = 1;
+
+#NODE H
+    #PARAMETERS
+        a = 2;
+        b = (a == 2);
+        c = {b, a == 3};
+    #VARIABLES
+        internal : x;
+    #CONSTRAINTS
+        CUSTOMFUNC(a == 2);
```

Comment:
For this to pass semantic checking, `CUSTOMFUNC` should be defined

Reply:
For now, the tests are mainly for parsing; we indeed ned to separate "things that should parse, but are not semantically sound" from "things that should parse and are semantically sound" but that's not the purpose of this PR